suitcase/tiff/__init__.py

```diff
@@ -1,25 +1,160 @@
-from ._version import get_versions
-__version__ = get_versions()['version']
-del get_versions
-
-
-# Suitcase subpackages must follow strict naming and interface conventions. The
+# suitcase subpackages must follow strict naming and interface conventions. The
 # public API should include some subset of the following. Any functions not
 # implemented should be omitted, rather than included and made to raise
 # NotImplementError, so that a client importing this library can immediately
 # know which portions of the suitcase API it supports without calling any
 # functions.
 #
-# def export(...)
-#     ...
-#
-#
-# def ingest(...):
-#     ...
-#
-#
-# def reflect(...):
-#     ...
-#
-#
-# handlers = []
+from collections import defaultdict
+from pathlib import Path
+import json
+import tifffile
+import event_model
+import numpy
+import suitcase
+from ._version import get_versions
+
+__version__ = get_versions()['version']
+del get_versions
+
+
+class SuitcaseTiffError(Exception):
+    ...
+
+
+class NonSupportedDataShape(SuitcaseTiffError):
+    '''used to indicate that non-supported data type is being saved
+    '''
+    ...
+
+
+def export(gen, directory, file_prefix='', **kwargs):
```

Comment:
`file_prefix` should probably be a format string that pulls from header to inject information into the file names.  As currently written, you can not sequentially export N headers without putting logic in the user code to generate unique file names.

We have presedence for this in https://nsls-ii.github.io/bluesky/callbacks.html#exporting-image-data-as-tiff-files

The default format should probably be `'{start[uid]}_'`.

Reply:
Let's consider what @tacaswell's suggestion of a default of `file_prefix={uid}` would make possible. As currently implemented, the following simple usage:

```py
export(db(since='2019-01-01'), '/media/my_usb_drive')
```

would error out when the second run is processed as it will attempt to overwrite the files from the first run. This may produce the impression that the exporter is "broken". If instead, the above usage produces UUID4-named files by default, the user (likely under guidance from BL staff or consultation with DAMA) can then make the names nicer in whatever way makes sense for their needs and their available metadata, as in:

```py
export(db(since='2019-01-01', '/media/my_usb_drive', file_prefix='{grad_student}-{temperature}-{etc}')
```

Would it be better for the "out-of-the-box" behavior be something that is technically guaranteed (by virtue of the uniqueness of UUID4) to produce files, albiet machine-named ones, rather than error out?